src/ClassDiscovery.php

```diff
@@ -109,7 +110,7 @@ private static function storeInCache($type, $class)
     /**
      * Set new strategies and clear the cache.
      *
-     * @param DiscoveryStrategy[] $strategies
+     * @param array $strategies string array of fully qualified class name to a DiscoveryStrategy
      */
     public static function setStrategies(array $strategies)
```

Comment:
a similar test could be done here, setting the strategies to only that test strategy that returns a mock client